### ISSUE TITLE
Add specific test for redirection scenario

### ISSUE THREAD

**Issue by Kartones** [CartoDB/cartodb#6081]
https://github.com/CartoDB/cartodb/pull/6080  This was caused due to wrong manual tests and no automated ones... fix that

### Scenario to reproduce desired flow:
- have current_user (a session)
- vis privacy public
- vis not shared with current_user especifically (else goes to show_organization_public_map)
- vis not owned by current_user
- vis is from organization user
- cartodb is not in subdomainless mode
- current_user is either not an org user or from another org (`CartoDB.subdomain_from_request(request) != user.organization.name`)
- not redirected (`!request.params[:redirected].present?`)

TL;DR:
- Org user should setup a public map
- Paste the public map url ommiting the organization name in a non-org user logged in session browser, e.g. `http://x.localhost.lan:3000/viz/3a998052-6c29-11e5-9f1a-080027221ad0/public_map` instead of `http://sampleorg.localhost.lan:3000/u/x/viz/3a998052-6c29-11e5-9f1a-080027221ad0/public_map`

**Comment by Kartones** [comment]
Branch: `6081-proper-redirect-fix`
